src/api/v1beta1/properties.go

```diff
@@ -137,6 +137,40 @@ func (dk *DynaKube) ActiveGateImage() string {
 	return fmt.Sprintf("%s/linux/activegate:latest", registry)
 }
 
+// EecImage returns the Extension Controller image to be used with the dk DynaKube instance.
+func (dk *DynaKube) EecImage() string {
+	if dk.ActiveGateMode() {
+		eecImage := dk.FeatureCustomEecImage()
+		if len(eecImage) > 0 {
+			return eecImage
+		}
+	}
+
+	if dk.Spec.APIURL == "" {
+		return ""
+	}
+
+	registry := buildImageRegistry(dk.Spec.APIURL)
+	return fmt.Sprintf("%s/linux/dynatrace-eec:latest", registry)
+}
+
+// StatsdImage returns the StatsD data source image to be used with the dk DynaKube instance.
+func (dk *DynaKube) StatsdImage() string {
```

Comment:
these methods are pretty similar, logic could be reused

Reply:
Looked like a bit of an overkill to me, but I don't have a strong opinion here so refactored it to get rid of the redundancy.